mm2src/lp_ordermatch.rs

```diff
@@ -3935,29 +3935,34 @@ struct HistoricalOrder {
 }
 
 impl HistoricalOrder {
-    fn build(new_order: &new_protocol::MakerOrderUpdated, old_order: &MakerOrder) -> HistoricalOrder {
-        HistoricalOrder {
-            max_base_vol: if new_order.new_max_volume.is_some() {
-                Some(old_order.max_base_vol.clone())
-            } else {
-                None
-            },
-            min_base_vol: if new_order.new_min_volume.is_some() {
-                Some(old_order.min_base_vol.clone())
-            } else {
-                None
-            },
-            price: if new_order.new_price.is_some() {
-                Some(old_order.price.clone())
-            } else {
-                None
-            },
-            updated_at: old_order.updated_at,
-            conf_settings: if new_order.conf_settings == old_order.conf_settings {
-                None
-            } else {
-                old_order.conf_settings
+    fn build(new_order: &new_protocol::MakerOrderUpdated, old_order: &MakerOrder) -> Result<HistoricalOrder, String> {
+        match new_order {
+            new_protocol::MakerOrderUpdated::V1(_) => {
+                return ERR!("HistoricalOrder build function is supported only for MakerOrderUpdated::V2")
```

Comment:
I think returning the error here is excessive as we can use
```rust
if let Some(settings) = new_order.new_conf_settings() {
    ...
}
```
that will return None in the case when V1 is passed occasionally.

Reply:
Just to clarify this code
```
conf_settings: if let Some(settings) = new_order.new_conf_settings() {
    if Some(settings) == old_order.conf_settings {
        None
    } else {
        old_order.conf_settings
    }
} else {
    None
},
```
the first if condition is used to check if V1 is used. the second is to check if update_maker_order was called with the same conf_settings as before so the conf_settings are not written in history as changed.